### PR TITLE
Crud for manufacturer

### PR DESCRIPTION
**Pull request by Claverin**

### CODE REVIEW

File: WineShop/Controllers/ManufacturerController.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult DeletePost(int? id)
+        {
+            var obj = _db.Manufacturer.Find(id);
+            if (obj == null)
+            {
+                return NotFound();
+            }
+            else
```
Comment (KamilGaluszka) [inline WineShop/Controllers/ManufacturerController.cs:97]: in if u have return, u don't need else

Comment (KamilGaluszka) [inline WineShop/Controllers/ManufacturerController.cs:97]: in if u have return, u don't need else

File: WineShop/Views/Manufacturer/Create.cshtml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                        </div>
+                        <div class="col">
+                             <a asp-action="Index" class="btn btn-success w-100">Back</a>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+
```
Comment (KamilGaluszka) [inline WineShop/Views/Manufacturer/Create.cshtml:34]: empty line

File: WineShop/Views/Manufacturer/Create.cshtml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                    </div>
+                    <div class="col-8">
+                        <input asp-for="Name" class="form-control" />
+                        <span asp-validation-for="Name" class="text-danger"></span>
+                    </div>
+
+                </div>
+                <div class="form-group row mt-3">
+                    <div class="col-8 offset-4 row">
+
```
Comment (KamilGaluszka) [inline WineShop/Views/Manufacturer/Create.cshtml:22]: empty line

File: WineShop/Views/Manufacturer/Create.cshtml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            <div class="col-8">
+                <div class="form-group row">
+                    <div class="col-4">
+                        <label asp-for="Name"></label>
+                    </div>
+                    <div class="col-8">
+                        <input asp-for="Name" class="form-control" />
+                        <span asp-validation-for="Name" class="text-danger"></span>
+                    </div>
+
```
Comment (KamilGaluszka) [inline WineShop/Views/Manufacturer/Create.cshtml:18]: empty line

File: WineShop/Views/Manufacturer/Delete.cshtml [hunk 5]
```diff
@@ -0,0 +1,43 @@
+﻿@model WineShop.Models.Manufacturer
+
+<form method="post" asp-action="DeletePost">
+    <input hidden type="number" value=@Model.ID_Manufacturer name="id" />
+    <div class="border p-3">
+
```
Comment (KamilGaluszka) [inline WineShop/Views/Manufacturer/Delete.cshtml:6]: empty line

File: WineShop/Views/Manufacturer/Delete.cshtml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        <div class="row">
+            <div class="col-8">
+                <div class="form-group row">
+                    <div class="col-4">
+                        <label asp-for="Name"></label>
+                    </div>
+                    <div class="col-8">
+                        <input asp-for="Name" disabled class="form-control" />
+                    </div>
+
```
Comment (KamilGaluszka) [inline WineShop/Views/Manufacturer/Delete.cshtml:19]: empty line

File: WineShop/Views/Manufacturer/Delete.cshtml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+                        </div>
+                        <div class="col">
+                            <a asp-action="Index" class="btn btn-success w-100">Back</a>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+
```
Comment (KamilGaluszka) [inline WineShop/Views/Manufacturer/Delete.cshtml:34]: empty line

File: WineShop/Views/Manufacturer/Delete.cshtml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+</form>
+
+@section Scripts{
+
```
Comment (KamilGaluszka) [inline WineShop/Views/Manufacturer/Delete.cshtml:38]: empty line

File: WineShop/Views/Manufacturer/Delete.cshtml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    @{
+        <partial name="_ValidationScriptsPartial" />
+    }
+
```
Comment (KamilGaluszka) [inline WineShop/Views/Manufacturer/Delete.cshtml:42]: empty line

File: WineShop/Views/Manufacturer/Edit.cshtml [hunk 10]
```diff
@@ -0,0 +1,44 @@
+﻿@model WineShop.Models.Manufacturer
+
+<form method="post">
+    <input asp-for="ID_Manufacturer" hidden />
+    <div class="border p-3">
+
```
Comment (KamilGaluszka) [inline WineShop/Views/Manufacturer/Edit.cshtml:6]: empty line

File: WineShop/Views/Manufacturer/Edit.cshtml [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+            <div class="col-8">
+                <div class="form-group row">
+                    <div class="col-4">
+                        <label asp-for="Name"></label>
+                    </div>
+                    <div class="col-8">
+                        <input asp-for="Name" class="form-control" />
+                        <span asp-validation-for="Name" class="text-danger"></span>
+                    </div>
+
```
Comment (KamilGaluszka) [inline WineShop/Views/Manufacturer/Edit.cshtml:20]: empty line

File: WineShop/Views/Manufacturer/Edit.cshtml [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+                        </div>
+                        <div class="col">
+                            <a asp-action="Index" class="btn btn-success w-100">Back</a>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+
```
Comment (KamilGaluszka) [inline WineShop/Views/Manufacturer/Edit.cshtml:35]: empty line

File: WineShop/Views/Manufacturer/Edit.cshtml [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+</form>
+
+@section Scripts{
+
```
Comment (KamilGaluszka) [inline WineShop/Views/Manufacturer/Edit.cshtml:39]: empty line

File: WineShop/Views/Manufacturer/Edit.cshtml [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    @{
+        <partial name="_ValidationScriptsPartial" />
+    }
+
```
Comment (KamilGaluszka) [inline WineShop/Views/Manufacturer/Edit.cshtml:43]: empty line

File: WineShop/Controllers/ManufacturerController.cs [hunk 15]
```diff
@@ -18,5 +18,88 @@ public IActionResult Index()
...
+        //POST - CREATE
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Create(Manufacturer obj)
+        {
+            if (ModelState.IsValid)
+            {
+                _db.Manufacturer.Add(obj);
+                _db.SaveChanges();
+                return RedirectToAction("Index");
```
Comment (MaciejBuczek) [inline WineShop/Controllers/ManufacturerController.cs:36]: do not use magic strings use nameof(Index) instead

File: WineShop/Controllers/ManufacturerController.cs [hunk 16]
```diff
@@ -18,5 +18,88 @@ public IActionResult Index()
...
+        //POST - EDIT
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Edit(Manufacturer obj)
+        {
+            if (ModelState.IsValid)
+            {
+                _db.Manufacturer.Update(obj);
+                _db.SaveChanges();
+                return RedirectToAction("Index");
```
Comment (MaciejBuczek) [inline WineShop/Controllers/ManufacturerController.cs:66]: do not use magic strings use nameof(Index) instead

**Review by MaciejBuczek — CHANGES_REQUESTED** [review]

**Pull request merged**